src/libYARP_sig/include/yarp/sig/Vector.h

```diff
@@ -257,6 +257,24 @@ class yarp::sig::VectorOf : public VectorBase
         len = 0;
         first = nullptr;
     }
+
+    /**
+    * Read vector from a connection.
+    * return true if a vector was read correctly
+    */
+    virtual bool read(yarp::os::ConnectionReader& connection) override
```

Comment:
What is the point in re-implementing these methods? If I understand it correctly, they just return the result of the parent's method, that is exactly what would happen if you don't re-implement them, but at the cost of an extra method call. Am I missing something?

Also I totally missed this change at my first review, this confirms that this kind of changes (a change to a well established class) does not belong to such a pull request (new feature), and it should be done in a separate PR.

Finally if such a change is required, there should be an entry in the release notes.

Reply:
Maybe it would be useful to specialize these functions for types for which we can provide compatibility with Bottle as it is done for Vector. To be discussed separately?